src/solver/rosenbrock.cu

```diff
@@ -125,7 +125,7 @@ namespace micm
           if (g_tid < n)
           {
             d_ymax = max(fabs(d_y_old[g_tid]), fabs(d_y_new[g_tid]));
-            d_scale = atol + rtol * d_ymax;
+            d_scale = atol[g_tid] + rtol * d_ymax;
```

Comment:
I think this won't work because `g_tid` will loop through the whole vector length, which will be `number_of_grid_cells * number_of_species` here. My understanding is that `atol` only has the size of `number_of_species`?

Reply:
Well. The number of tolerances is the number of species, but then I duplicated those across all grid cells. So if we have three species with tolerances [(a, 0.1), (b, 0.2), (c, 0.3)] and are solving 3 grid cells, then the actual atol array that is passed in (or will be once I add it) will be [0.1, 0.2, 0.3, 0.1, 0.2, 0.3, 0.1, 0.2, 0.3]

So. I'm not sure what that means. This also means that we are ballooning memory on the CPU when we don't need to, but it seems like this thing could be beneficial for the GPU to reduce cache misses.

Either way, I need to get a release of the python code using these tolerances out to A&M before tomorrow. Should I just make an issue for this specific thing?

Reply:
Thanks Kyle. If `atol` is duplicated for all the grid cells, then there is no problem with this modification. But if we want to save memory for `atol`, we will need `g_tid % number_of_species` instead here.